modules/fbs/includes/classes/FBS.php

```diff
@@ -87,10 +90,17 @@ class FBS {
    *   FBSAuthenticationHandler.
    * @param Serializer|null $serializer
    *   Serializer to use. Defaults to JsonMapperSerializer instance.
+   *
+   * @throws FBSException
+   *   If provider agency id or endpoint are not set.
    */
-  public  function __construct($agency_id, $endpoint, HttpClient $client = NULL, Serializer $serializer = NULL) {
+  public function __construct($agency_id, $endpoint, HttpClient $client = NULL, Serializer $serializer = NULL) {
+    if (empty($agency_id) || empty($endpoint)) {
+      throw new FBSException('Provider agency id or endpoint are not set.');
```

Comment:
No need for a specific FBSException here. Client code should catch Exception anyway, so it's quite sufficient to throw a RuntimeException.